### DISCUSSION TITLE
Page loading progress bar (nprogress)

### DISCUSSION THREAD

**Discussion by elmaxe** [mantinedev/mantine#1053] [Feature requests]
### Usage example, including component/hook API

Can be used to indicate that something is loading. See for example [nprogress](https://www.npmjs.com/package/nprogress) or [react-top-loading-bar](https://www.npmjs.com/package/react-top-loading-bar)

### Possible implementation – describe how the feature can be implemented

Should probably use the existing `Progress` component when implementing this.

I'm not 100% sure what the best way to implement this would be. Perhaps a hook:

```tsx
const App = () => {
  const { start, stop, increment, ProgressBar }= useProgressBar({
    color: "pink",
  })

  return (
    <div>
      <ProgressBar />
      <Button onClick={start}>Load</Button>
      <Button onClick={end}>Finish</Button>
    </div>
  )
}
```

### Do you want to contribute this feature and create a pull request

Yes

**Comment by rtivital** [comment]
Thanks for feature request, it's a good feature that should be added as a new package `@mantine/nprogress`, with this implementation:

- Provider component
```tsx
import { NProgressProvider } from '@mantine/nprogress';
import { MantineProvider } from '@mantine/core';

export function App() {
  return (
    <MantineProvider>
      <NProgressProvider color="red" size={15}>
        <AppItself />
      </NProgressProvider>
    </MantineProvider>
  );
}
```

- consumer component

```tsx
const Consumer = () => {
  // hook can override options defined on Provider, options are not required
  const { start, stop, set }= useNProgress({ size: 5, color: 'blue' });

  return (
    <>
      <Button onClick={start}>Load</Button>
      <Button onClick={() => set(30)}>Set 30</Button>
      <Button onClick={end}>Finish</Button>
    </>
  )
}
```

Do you want to work on this package?

**Comment by elmaxe** [comment]
@rtivital 

Yes that would be fun!

**Comment by rtivital** [comment]
Alright then, submit a PR, and don't hesitate to reach out on Discord in “Contributing” channel if you have any questions

**Comment by elmaxe** [comment]
Will do!

**Comment by albertbronsky** [comment]
@elmaxe @rtivital Any _progress_ (no pun intended) on this? :)

**Comment by elmaxe** [comment]
> @elmaxe @rtivital Any _progress_ (no pun intended) on this? :)

I started a bit. I however realised that I won't have the time to implement this.

**Comment by jerebtw** [comment]
> > @elmaxe @rtivital Any _progress_ (no pun intended) on this? :)
> 
> I started a bit. I however realised that I won't have the time to implement this.

If you are okay with it, I would like to try to implement it.
